vm/src/types/slot.rs

```diff
@@ -900,6 +932,40 @@ pub trait AsSequence: PyPayload {
     }
 }
 
+#[pyimpl]
+pub trait AsNumber: PyPayload {
+    #[inline]
+    #[pyslot]
+    fn slot_as_number(zelf: &PyObject, vm: &VirtualMachine) -> Cow<'static, PyNumberMethods> {
```

Comment:
I think for heap object we need to use Cow::Owned witch implemented in as_number_wrapper

Reply:
They are not decided by object, but decided by type. Then I think the type can hold a cloned `PyNumberMethods` with edited values. instead of `Cow` for every object when `as_number` is called.

Reply:
Yes, it should be stored in the type struct, but we can fix it later